sky/utils/kubernetes_utils.py

```diff
@@ -1016,12 +1017,17 @@ def fill_ssh_jump_template(ssh_key_secret: str, ssh_jump_image: str,
 
 
 def check_port_forward_mode_dependencies() -> None:
-    """Checks if 'socat' is installed"""
+    """Checks if 'socat', 'netcat', 'shuf(coreutils)' are installed"""
     # We store the dependency list as a list of lists. Each inner list
     # contains the name of the dependency, the command to check if it is
     # installed, and the package name to install it.
+    osused = pltsys()
     dependency_list = [['socat', ['socat', '-V'], 'socat'],
                        ['nc', ['nc', '-h'], 'netcat']]
+    if osused == 'Darwin':
+        dependency_list.append(['gshuf', ['gshuf', '--version'], 'coreutils'])
+    elif osused == 'Linux':
+        dependency_list.append(['shuf', ['shuf', '--version'], 'coreutils'])
```

Comment:
Similar to above, can we first check for shuf, then fallback to gshuf if its not present? That way we don't need OS-specific checks.

Reply:
In that case dependency list can have only shuf and not bother about gshuf? I had it this way caz, in future if we have any specific package differences between each OS, we can add it to this block.